client/src/components/EditFlagForm.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from "react";
+import { useParams } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+import { editFlag } from "../actions/flagActions";
+
+const EditFlagForm = ({ setIsEditing }) => {
+  const dispatch = useDispatch();
+  const { flagId } = useParams();
+  const flagData = useSelector((state) =>
+    state.flags.find((flag) => flag.id === +flagId)
+  );
+
+  const [newName, setNewName] = useState(flagData ? flagData.name : "");
+  const [newDescription, setNewDescription] = useState(
+    flagData ? flagData.description : ""
+  );
+  const [newPercent, setNewPercent] = useState(
+    flagData ? flagData.percentage_split : 100
+  );
+
+  const handleSaveEdits = () => {
```

Comment:
This is the source of the new bug I just mentioned. You need to pass in `e` here and run `e.preventDefault()` in the first line.